Modules/Loadable/Markups/MRML/vtkCurveMeasurementsCalculator.cxx

```diff
@@ -0,0 +1,352 @@
+  double prevMeanPoint[3] = {currPoint[0], currPoint[1], currPoint[2]}; // pm (Skip first point)
+  double kappa = 0.0;
+  double currentLength = 0.0;
+  double length = 0.0;
+
+  curvatureValues->InsertValue(linePoints->GetId(0), 0.0); // The curvature for the first cell is 0.0
+
+  prevPoint[0] = currPoint[0]; prevPoint[1] = currPoint[1]; prevPoint[2] = currPoint[2];
+  for (vtkIdType idx=1; idx<numberOfPoints-1; ++idx)
+    {
+    currPoint = points->GetPoint(linePoints->GetId(idx+1));
+
+    diffVector[0] = currPoint[0]-prevPoint[0];
+    diffVector[1] = currPoint[1]-prevPoint[1];
+    diffVector[2] = currPoint[2]-prevPoint[2];
+    diffNorm = sqrt(diffVector[0]*diffVector[0] + diffVector[1]*diffVector[1] + diffVector[2]*diffVector[2]);
+
+    normDiffVector[0] = diffVector[0] / diffNorm;
+    normDiffVector[1] = diffVector[1] / diffNorm;
+    normDiffVector[2] = diffVector[2] / diffNorm;
+
+    // Local curvature
+    kappa = sqrt( (normDiffVector[0]-prevNormDiffVector[0])*(normDiffVector[0]-prevNormDiffVector[0])
+                + (normDiffVector[1]-prevNormDiffVector[1])*(normDiffVector[1]-prevNormDiffVector[1])
+                + (normDiffVector[2]-prevNormDiffVector[2])*(normDiffVector[2]-prevNormDiffVector[2]) )
+            / diffNorm;
+    curvatureValues->InsertValue(linePoints->GetId(idx), kappa);
+
+    // Statistics
+    meanPoint[0] = (currPoint[0]+prevPoint[0]) / 2.0;
+    meanPoint[1] = (currPoint[1]+prevPoint[1]) / 2.0;
+    meanPoint[2] = (currPoint[2]+prevPoint[2]) / 2.0;
+
+    currentLength = sqrt( (meanPoint[0]-prevMeanPoint[0])*(meanPoint[0]-prevMeanPoint[0])
+                        + (meanPoint[1]-prevMeanPoint[1])*(meanPoint[1]-prevMeanPoint[1])
+                        + (meanPoint[2]-prevMeanPoint[2])*(meanPoint[2]-prevMeanPoint[2]) );
+    if (kappa < minKappa)
+      {
+      minKappa = kappa;
+      }
+    else if (kappa > maxKappa)
+      {
+      maxKappa = kappa;
+      }
+    meanKappa += kappa * currentLength; // weighted mean
+    length += currentLength;
+
+    // Propagate current values to previous
+    for (int i=0; i<3; ++i)
+      {
+      prevPoint[i] = currPoint[i];
+      prevMeanPoint[i] = meanPoint[i];
+      prevNormDiffVector[i] = normDiffVector[i];
+      }
+    } // For each line point
+
+  // Final point
+  curvatureValues->InsertValue(linePoints->GetId(numberOfPoints-1), 0.0); // The curvature for the last cell is 0.0
+
+  currentLength = sqrt( (prevPoint[0]-prevMeanPoint[0])*(prevPoint[0]-prevMeanPoint[0])
+                      + (prevPoint[1]-prevMeanPoint[1])*(prevPoint[1]-prevMeanPoint[1])
+                      + (prevPoint[2]-prevMeanPoint[2])*(prevPoint[2]-prevMeanPoint[2]) );
+  length += currentLength;
+  meanKappa = meanKappa / length;
+
+  // Set curvature array to output
+  polyData->GetPointData()->AddArray(curvatureValues);
+  polyData->GetPointData()->SetActiveScalars("Curvature"); //TODO: Do it here or in displayable manager when turning off curvature visibility?
```

Comment:
Multiple measurements may produce arrays, so probably we should not set the active scalars here.